File: code/Project_4/BT5/c++/problems/problem_1_1.cpp
#include <iostream>
using namespace std;

int complete_graph_edges(int n) {
    if (n <= 1) return 0;
    return n * (n - 1) / 2;
}

int complete_bipartite_edges(int p, int q) {
    return p * q;
}

int main() {
    cout << "Complete Graph Kn:" << endl;
    for (int n = 1; n <= 5; n++) {
        cout << "K" << n << ": " << complete_graph_edges(n) << " edges" << endl;
    }
    
    cout << "\nComplete Bipartite Graph Kp,q:" << endl;
    int test_cases[][2] = {{1,1}, {2,2}, {2,3}, {3,4}};
    for (int i = 0; i < 4; i++) {
        int p = test_cases[i][0];
        int q = test_cases[i][1];
        cout << "K" << p << "," << q << ": " << complete_bipartite_edges(p, q) << " edges" << endl;
    }
    
    return 0;
}

File: code/Project_4/BT5/python/prolems/problem_1_1.py
def complete_graph_edges(n):
    """Tính số cạnh của đồ thị đầy đủ Kn"""
    if n <= 1:
        return 0
    return n * (n - 1) // 2

def complete_bipartite_edges(p, q):
    """Tính số cạnh của đồ thị hai phía đầy đủ Kp,q"""
    return p * q

def main():

    print("Complete Graph Kn:")
    for n in [1, 2, 3, 4, 5]:
        edges = complete_graph_edges(n)
        print(f"K{n}: {edges} edges")
    
    print("\nComplete Bipartite Graph Kp,q:")
    test_cases = [(1,1), (2,2), (2,3), (3,4)]
    for p, q in test_cases:
        edges = complete_bipartite_edges(p, q)
        print(f"K{p},{q}: {edges} edges")

if __name__ == "__main__":
    main()